WordPress/Classes/ViewRelated/Blaze Campaigns/BlazeCampaignTableViewCell.swift

```diff
@@ -0,0 +1,196 @@
+        stackView.spacing = Metrics.detailsOuterStackViewSpacing
+        stackView.addArrangedSubviews([detailsInnerStackView, featuredImageView])
+        return stackView
+    }()
+
+    private lazy var detailsInnerStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .vertical
+        stackView.spacing = Metrics.detailsInnerStackViewSpacing
+        stackView.addArrangedSubviews([titleLabel, statsStackView])
+        return stackView
+    }()
+
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.adjustsFontForContentSizeCategory = true
+        label.adjustsFontSizeToFitWidth = true
+        label.font = WPStyleGuide.fontForTextStyle(.subheadline, fontWeight: .semibold)
+        label.numberOfLines = 0
+        label.textColor = .text
+        return label
+    }()
+
+    private lazy var statsStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .horizontal
+        stackView.spacing = Metrics.statsStackViewSpacing
+        return stackView
+    }()
+
+    private lazy var featuredImageView: CachedAnimatedImageView = {
+        let imageView = CachedAnimatedImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+
+        NSLayoutConstraint.activate([
+            imageView.widthAnchor.constraint(equalToConstant: Metrics.featuredImageSize),
+            imageView.heightAnchor.constraint(equalTo: imageView.widthAnchor)
+        ])
+
+        imageView.clipsToBounds = true
+        imageView.layer.cornerRadius = Metrics.featuredImageCornerRadius
+        return imageView
+    }()
+
+    // MARK: - Properties
+
+    private lazy var imageLoader: ImageLoader = {
+        return ImageLoader(imageView: featuredImageView, gifStrategy: .mediumGIFs)
+    }()
+
+    // MARK: - Initializers
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        commonInit()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        commonInit()
+    }
+
+    // MARK: - Public
+
+    func configure(with viewModel: BlazeCampaignViewModel, blog: Blog) {
+        statusView.configure(with: viewModel.status)
+
+        titleLabel.text = viewModel.title
+
+        imageLoader.prepareForReuse()
+        featuredImageView.isHidden = viewModel.imageURL == nil
+        if let imageURL = viewModel.imageURL {
+            let host = MediaHost(with: blog, failure: { error in
+                WordPressAppDelegate.crashLogging?.logError(error)
+            })
+
+            let preferredSize = CGSize(width: featuredImageView.frame.width, height: featuredImageView.frame.height)
```

Comment:
It gets called before the layout pass. The frames are zeros:

```swift
(lldb) po featuredImageView.frame
▿ (0.0, 0.0, 0.0, 0.0)
  ▿ origin : (0.0, 0.0)
    - x : 0.0
    - y : 0.0
  ▿ size : (0.0, 0.0)
    - width : 0.0
    - height : 0.0
```

Reply:
Fixed! [eaa3395](https://github.com/wordpress-mobile/WordPress-iOS/pull/20886/commits/eaa339516bd092f90db16175e6054f91ceef1589)